trovi/models.py

```diff
@@ -70,8 +167,43 @@ def access_count(self) -> int:
         """
         return self.events.filter(event_type=ArtifactEvent.EventType.LAUNCH).count()
 
+    @staticmethod
+    def generate_slug(instance: "ArtifactVersion", created: bool = False, **_):
+        """
+        Generates a slug in the format of YYYY-MM-DD((.#)?) where ".#"
+        is an index starting at 1 which increments automatically for each version
+        published on the same given day.
 
-class ArtifactEvent(models.Model):
+        The slug is stored in the instance's slug field and saved
+        """
+        if created:
+            time_stamp = instance.created_at.strftime("%Y-%m-%d")
+            versions_today = ArtifactVersion.objects.filter(
+                artifact__created_at__year=instance.created_at.year,
```

Comment:
This is simpler, I think equivalent, and reduces from 3 kwargs to one:

```suggestion
                artifact__created_at__date=instance.created_at.date()
```